server/src/main/java/org/elasticsearch/action/bulk/BulkRequest.java

```diff
@@ -84,6 +84,7 @@ public class BulkRequest extends ActionRequest
     private String globalIndex;
     private Boolean globalRequireAlias;
     private Boolean globalRequireDatsStream;
+    private boolean includeSourceOnError = true;
```

Comment:
nit: it would be more clear if the default were set in the StreamInput ctor in the else case

Reply:
If it was only for the single StreamInput ctor, I'd agree. Though, there's two additional ctors used in various places. I want to be sure the default is always correctly set.

Reply:
made it a nullable Boolean to avoid having to handle defaults in multiple places, with that this should also be resolved
https://github.com/elastic/elasticsearch/pull/120725/commits/6a3e74a53bd4cf01cf75c7f59cc5f8ce1be29aef